operator/internal/deploy/chart_manager.go

```diff
@@ -0,0 +1,222 @@
+		return nil, fmt.Errorf("error listing watcher resources: %w", err)
+	}
+	if len(watcherList.Items) == 0 {
+		return nil, ErrFoundZeroWatchers
+	}
+	chartCfg := generateWatchableConfigs(watcherList)
+	bytes, err := k8syaml.Marshal(chartCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	kcpAddr, err := m.resolveKcpAddr(ctx, kcpClient)
+	if err != nil {
+		return nil, err
+	}
+	return map[string]interface{}{
+		"triggerLabel":          time.Now().Format(triggerLabelTimeFormat),
+		"kcpAddr":               kcpAddr,
+		"resourcesLimitsMemory": m.config.SkrWebhookMemoryLimits,
+		"resourcesLimitsCPU":    m.config.SkrWebhookCPULimits,
+		customConfigKey:         string(bytes),
+	}, nil
+}
+
+func (m *EnabledSKRWebhookChartManager) resolveKcpAddr(ctx context.Context, kcpClient client.Client) (string, error) {
+	if m.kcpAddr != "" {
+		return m.kcpAddr, nil
+	}
+	// Get external IP from the ISTIO load balancer external IP
+	loadBalancerService := &corev1.Service{}
+	if err := kcpClient.Get(ctx, client.ObjectKey{Name: IngressServiceName, Namespace: IstioSytemNs},
+		loadBalancerService); err != nil {
+		return "", err
+	}
+	if len(loadBalancerService.Status.LoadBalancer.Ingress) == 0 {
+		return "", ErrLoadBalancerIPIsNotAssigned
+	}
+	externalIP := loadBalancerService.Status.LoadBalancer.Ingress[0].IP
+	var port int32
+	for _, loadBalancerPort := range loadBalancerService.Spec.Ports {
+		if loadBalancerPort.Name == "http2" {
+			port = loadBalancerPort.Port
+			break
+		}
+	}
+	m.kcpAddr = net.JoinHostPort(externalIP, strconv.Itoa(int(port)))
+	return m.kcpAddr, nil
+}
+
+func generateWatchableConfigs(watcherList *v1alpha1.WatcherList) map[string]WatchableConfig {
+	chartCfg := make(map[string]WatchableConfig, len(watcherList.Items))
+	for _, watcher := range watcherList.Items {
+		statusOnly := watcher.Spec.Field == v1alpha1.StatusField
+		chartCfg[watcher.GetModuleName()] = WatchableConfig{
+			Labels:     watcher.Spec.LabelsToWatch,
+			StatusOnly: statusOnly,
+		}
+	}
+	return chartCfg
+}
+
+func installOrRemoveChartOnSKR(ctx context.Context, deployInfo modulelib.InstallInfo, mode Mode,
+	enableWebhookPreInstallCheck bool,
+) error {
+	logger := logf.FromContext(ctx)
+	if mode == ModeUninstall {
+		uninstalled, err := modulelib.UninstallChart(&logger, deployInfo, nil, nil)
+		if err != nil {
+			return fmt.Errorf("failed to uninstall webhook config: %w", err)
+		}
+		if !uninstalled {
+			return ErrSKRWebhookWasNotRemoved
+		}
+
+		return nil
+	}
+	if enableWebhookPreInstallCheck {
+		// TODO(khlifi411): verify webhook configuration with watchers' configuration before re-installing the chart
+		ready, err := modulelib.ConsistencyCheck(&logger, deployInfo, nil, nil)
+		if err != nil {
+			return fmt.Errorf("failed to verify webhook resources: %w", err)
+		}
+		if ready {
+			logger.V(1).Info("chart resources already installed, nothing to do!")
+			return nil
+		}
+	}
+	installed, err := modulelib.InstallChart(&logger, deployInfo, nil, nil)
```

Comment:
You are not passing a chart cache here and this will cause performance issues during rendering

Reply:
Agree! This should be fixed when chart cache becomes optional, right?

Reply:
Not sure, I think chart cache should always be used. Please correct me here if wrong @adityabhatia